helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1928,10 +1928,7 @@ public ZNRecord update(ZNRecord currentData) {
         }
 
         ClusterConfig clusterConfig = new ClusterConfig(currentData);
-        Map<String, String> disabledInstances = new TreeMap<>();
-        if (clusterConfig.getDisabledInstances() != null) {
-          disabledInstances.putAll(clusterConfig.getDisabledInstances());
-        }
```

Comment:
If we remove this logic, we may lose the meaning of using updater.

Reply:
Could you clarify? I don't think any logic is removed from here.

Reply:
It's properly handled, see https://github.com/apache/helix/pull/2032/files#diff-8af2a416d09c30beedc9f110edd51472c0331df5c3c46225fdd3901b3cdf6f5e, it's changed to non-null.